File: SharedExample/app/src/main/java/com/example/sharedexample/MainActivity.java
package com.example.sharedexample;

import androidx.appcompat.app.AppCompatActivity;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    EditText et_save;
    String shared = "file";
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        et_save = (EditText)findViewById(R.id.et_save);
        SharedPreferences sharedPreferences = getSharedPreferences(shared, 0);
        String value = sharedPreferences.getString("android","");
        et_save.setText(value);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();

        SharedPreferences sharedPreferences = getSharedPreferences(shared,0);
        SharedPreferences.Editor editor = sharedPreferences.edit();
        String value = et_save.getText().toString();
        editor.putString("android",value);
        editor.commit();
//생명주기에 대한 이해필요, SharedPreferences 와 Editor에 대한 개념 보통
        // EditText안의 값을 빼올때는 String 형태의 변수를 이용해서 값을 꺼내옴.
        // js나 css처럼 마찬가지로 ID태그를 이용함.
    }
}

File: loginlecture/app/src/main/java/com/example/loginlecture/MainActivity.kt
package com.example.loginlecture

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    var existBlank = false
    var ispwSame = false
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        login_button.setOnClickListener{

            //EditText로부터 입력된 값을 받아온다.
            var id = login_id.text.toString()
            var pw = login_pw.text.toString()

            //쉐어드로부터 저장된 id,pw 값을 가져오기
            val sharedPreference = getSharedPreferences("")
        }


    }
}